Sources/AsyncKit/EventLoopFuture/Future+Collection.swift

```diff
@@ -128,4 +119,50 @@ extension EventLoopFuture where Value: Sequence {
             return try sequence.compactMap(transform)
         }
     }
+
+    /// A variant form of `flatMapEach(on:_:)` which guarantees:
+    ///
+    /// 1) Explicitly sequential execution of each future returned by the mapping
+    ///    closure; the next future does not being executing until the previous one
+    ///    has yielded a success result.
+    ///
+    /// 2) No further futures will be even paritially executed if any one future
+    ///    returns a failure result.
+    ///
+    /// Neither of these are provided by the original version of the method.
+    public func sequencedFlatMapEach<Result>(_ transform: @escaping (_ element: Value.Element) -> EventLoopFuture<Result>) -> EventLoopFuture<[Result]> {
+        var results: [Result] = []
+        
+        return self.flatMap {
+            $0.reduce(self.eventLoop.future()) { fut, elem in
+                fut.flatMap { transform(elem).map { results.append($0) } }
+            }
+        }.transform(to: results)
+    }
+
+    /// An overload of `sequencedFlatMapEach(_:)` which returns a `Void` future instead
+    /// of `[Void]` when the result type of the transform closure is `Void`.
+    public func sequencedFlatMapEach(_ transform: @escaping (_ element: Value.Element) -> EventLoopFuture<Void>) -> EventLoopFuture<Void> {
+        return self.flatMap {
```

Comment:
Generally speaking, I omit `return` for single-line statements, and for nested closures whose scoping is visually well-defined. I wouldn't claim to have any really "concrete" rules about it, though; it's one of items in my code style that I tend to do by instinct, case-by-case, and sometimes in different moods my choices change. That being said, I usually don't omit it for public methods, especially ones with documentation comments; I suppose I consider that the code equivalent of "formal prose", more or less 😅. In that context I'll always tend towards more verbose and explicitly expressive style than what is otherwise my strong tendency towards minimalism (often at the extreme expense of clarity 😰).